hopp/tools/design/wind/turbine_library_interface_tools.py

```diff
@@ -0,0 +1,200 @@
+        cp_curve = np.where(cp_curve<0,0,cp_curve)
+        cp_curve = list(cp_curve)
+
+        power_curve_kw = curve_tools.calculate_power_from_cp(wind_speeds,cp_curve,turbine_specs["rotor_diameter"],turbine_specs["rated_power"])
+    elif "power_kw" in turbine_curve_cols and "cp" not in turbine_curve_cols:
+        power_curve_kw = np.array(turbine_specs["power_curve"]["power_kw"].to_list())
+        power_curve_kw = np.where(power_curve_kw<0,0,power_curve_kw)
+        power_curve_kw = np.where(power_curve_kw>turbine_specs["rated_power"],turbine_specs["rated_power"],power_curve_kw)
+        power_curve_kw = list(power_curve_kw)
+        cp_curve = curve_tools.calculate_cp_from_power(wind_speeds,power_curve_kw)
+    else:
+        raise UserWarning(f"turbine {turbine_name} does not have minimum required power curve data (needs either power_kw or cp)")
+    if "ct" in turbine_specs["power_curve"].columns.to_list():
+        ct = turbine_specs["power_curve"]["ct"].to_list()
+    else:
+        ct = curve_tools.estimate_thrust_coefficient(wind_speeds,cp_curve)
+    
+    _, cp_curve = curve_tools.pad_power_curve(wind_speeds,cp_curve)
+    _, ct = curve_tools.pad_power_curve(wind_speeds,ct)
+    wind_speeds, power_curve_kw = curve_tools.pad_power_curve(wind_speeds,power_curve_kw)
+    if model_name == "floris":
+        power_thrust_table = {
+            # "ref_air_density": 1.225,
+            # "ref_tilt": 5.0,
+            # "cosine_loss_exponent_yaw": 1.88,
+            # "cosine_loss_exponent_tilt": 1.88,
+            "wind_speed":wind_speeds,
+            "power":power_curve_kw,
+            "thrust_coefficient":ct,
+            }
+    elif model_name == "pysam":
+        power_thrust_table = {
+            "wind_turbine_max_cp": max(cp_curve),
+            "wind_turbine_ct_curve":ct,
+            "wind_turbine_powercurve_windspeeds":wind_speeds,
+            "wind_turbine_powercurve_powerout":power_curve_kw,
+            }
+    return power_thrust_table
+
+
+def check_hub_height(turbine_specs,wind_plant):
+    """Check the hub-height from the turbine-library specs against the other possible hub-height entries.
+        If multiple hub-height options are available from the turbine_specs, it will choose one based on user-input information.
+
+    Args:
+        turbine_specs (dict): turbine specs loaded from turbine-models library.
+        wind_plant (:obj:`hopp.simulation.technologies.wind.floris.Floris` | :obj:`hopp.simulation.technologies.wind.wind_plant.WindPlant`): wind 
+            plant object for either PySAM or FLORIS wind simulation model.
+
+    Returns:
+        float: hub-height to use in meters.
+    """
+    turbine_name = turbine_specs["name"]
+    # if multiple hub height options are available
+    if isinstance(turbine_specs["hub_height"],list):
+        # check for hub height in wind_plant
+        if isinstance(wind_plant,object) and wind_plant is not None:
```

Comment:
We should be checking for one of the two options that are specified in the docstring, not `object`. Once this is done, the check for `is not None` can be removed.